### PR TITLE
Get & set options

### PR DESCRIPTION
**Pull request by scottsfarley93**
This PR fixes https://github.com/mapbox/mapbox-gl-geocoder/issues/176 by adding getter/setter methods.

This PRs get/set methods for the following options:
- `language`
- `zoom`
- `flyTo`
- `placeholder`
- `bbox`
- `countries`
- `types`
- `minLength`
- `limit`
- `filter`
- `proximity` (already implemented)

It's an open question as to whether we should support `get`, and particularly `set`, methods for these options, since changing their behavior more fundamentally alters the behavior of the plugin -- and I'm not sure of the use case of changing them after creating an instance with different values:

- `accessToken`
- `origin`
- `localGeocoder`
- `reverseMode`
- `reverseGeocode`

#208 will add get/set functions for html rendering functions and we should add analogous methods in #219 to support new options that we're adding in version 4 (#197)

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] run `npm run docs` and commit changes to API.md
 - [x] update CHANGELOG.md with changes under `master` heading before merging

\cc @katydecorah @yuletide

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+   * Get the zoom level the map will move to when there is no bounding box on the selected result
+   * @returns {Number} the map zoom
+   */
+  getZoom: function(){
+    return this.options.zoom;
+  },
+
+  /**
+   * Set the zoom level 
+   * @param {Number} zoom On geocoded result what zoom level should the map animate to when a `bbox` isn't found in the response. If a `bbox` is found the map will fit to the `bbox`.
```
Comment (katydecorah) [inline lib/index.js]: Small copy suggestion:

```suggestion
   * @param {Number} zoom The zoom level that the map should animate to when a `bbox` isn't found in the response. If a `bbox` is found the map will fit to the `bbox`.
```

File: lib/index.js [hunk 2]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+  /**
+   * Get the parameters used to fly to the selected response, if any
+   * @returns {MapboxGeocoder} this
+   */
+  getFlyTo: function(){
+    return this.options.flyTo;
+  },
+
+  /**
+   * Sets the flyTo options
```
Comment (katydecorah) [inline lib/index.js]: Small edit for consistency:

```suggestion
   * Set the flyTo options
```

File: lib/index.js [hunk 3]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+   * @returns {MapboxGeocoder} this
+   */
+  setPlaceholder: function(placeholder){
+    this.placeholder = (placeholder) ? placeholder : this._getPlaceholderText();
+    this._inputEl.placeholder = this.placeholder;
+    return this
+  },
+
+  /**
+   * Gets the bounding box used by the plugin
```
Comment (katydecorah) [inline lib/index.js]: For consistency:
```suggestion
   * Get the bounding box used by the plugin
```

File: lib/index.js [hunk 4]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+   * @returns {Array<Number>} the bounding box, if any
+   */
+  getBbox: function(){
+    return this.options.bbox;
+  },
+
+  /**
+   * Sets the bounding box to limit search results to
```
Comment (katydecorah) [inline lib/index.js]:
```suggestion
   * Set the bounding box to limit search results to
```

File: lib/index.js [hunk 5]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+  /**
+   * Get a list of the types to limit search results to 
+   * @returns {String} a comma separated list of types to limit to 
+   */
+  getTypes: function(){
+    return this.options.types;    
+  },
+
+  /**
+   * Set the countries to limit search results to
```
Comment (katydecorah) [inline lib/index.js]: Should this be types?

```suggestion
   * Set the types to limit search results to
```

File: lib/index.js [hunk 6]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+   * @param {String} countries a comma separated list of countries to limit to
```
Comment (katydecorah) [inline lib/index.js]: Should this also be types?

```suggestion
   * @param {String} types a comma separated list of countries to limit to 
```

File: lib/index.js [hunk 7]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+   * @returns {MapboxGeocoder} this
+   */
+  setMinLength: function(minLength){
+    this.options.minLength = minLength;
+    if (this._typeahead)  this._typeahead.minLength = minLength;
+    return this;
+  },
+
+  /**
+   * Get the limit value used by the plugin
```
Comment (katydecorah) [inline lib/index.js]:
```suggestion
   * Get the limit value for the number of results to display used by the plugin
```

File: lib/index.js [hunk 8]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+   * @param {String} countries a comma separated list of countries to limit to 
+   * @returns {MapboxGeocoder} this
+   */
+  setTypes: function(types){
+    this.options.types = types;
+    return this;
+  },
+
+  /**
+   * Get the minimum length used in the plugin
```
Comment (katydecorah) [inline lib/index.js]: Maybe call out what the minimum length refers to?

```suggestion
   * Get the minimum number of characters typed to trigger results used in the plugin
```

File: lib/index.js [hunk 9]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+   * @returns {Number}
+   */
+  getMinLength: function(){
+    return this.options.minLength;
+  },
+
+  /**
+   * Set the minimum length value used by the plugin
```
Comment (katydecorah) [inline lib/index.js]:
```suggestion
   * Set the minimum number of characters typed to trigger results used by the plugin
```

File: lib/index.js [hunk 10]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+  /**
+   * Get the limit value used by the plugin
+   * @returns {Number} the limit value
+   */
+  getLimit: function(){
+    return this.options.limit;
+  },
+
+  /**
+   * Set the limit value used by the plugin
```
Comment (katydecorah) [inline lib/index.js]:
```suggestion
   * Set the limit value for the number of results to display used by the plugin
```

File: lib/index.js [hunk 11]
```diff
@@ -447,33 +469,185 @@ MapboxGeocoder.prototype = {
...
+   * @returns {MapboxGeocoder} 
+   */
+  setLimit: function(limit){
+    this.options.limit = limit;
+    if (this._typeahead) this._typeahead.options.limit = limit;
+    return this;
+  },
+
+  /**
+   * Get the filter function used by the plugin
```
Comment (katydecorah) [inline lib/index.js]: I'm not exactly sure what `filter` does in suggestions, maybe we could add a little more to add context?

Comment (andrewharvey) [reply lib/index.js]: It filters the results shows :wink: See https://docs.mapbox.com/mapbox-gl-js/example/mapbox-gl-geocoder-limit-region/

**Pull request merged**